share/functions/__fish_complete_mysql.fish

```diff
@@ -0,0 +1,24 @@
+function __fish_complete_mysql
+    set -l command $argv[1]
```

Comment:
This isn't used anywhere.

Reply:
It is used by the `mysql` and `mysqldump` completions. I cannot put them both in the `mysql` file, otherwise the `mysqldump` completion will not work.
At least, that is what I understand of it, and it matches my experience (I initially had everything in one file).

Reply:
Aaah, okay, it's used down with `complete`. Sorry, I missed that. That's what happens when you try to make a quick pass during break, I guess.

Still I'd prefer if the `complete` calls happened in the complete files, so please just put `complete -c mysql ... -a '(__fish_complete_mysql_databases)` in mysql.fish, and with mysqldump in mysqldump.fish.

We've had a few completions that try to do this thing, and it's always a bit of a mess because it makes it hard to see where the options and conditions and such are defined.

Reply:
If I put `complete -c mysql ... -a '(__fish_complete_mysql_databases)` in the `completions/mysql.fish` file, I need to split `__fish_complete_mysql_databases` and `__fish_mysql_query` into separate files within `functions/`.
That might also make things less clear, let me know what you prefer.